File: HJ40 统计字符.cpp
#include <iostream>
using namespace std;

 int main()
 {
     string s;
     while(getline(cin, s))
     {
         int ch = 0;
         int num = 0;
         int space = 0;
         int other = 0;
         for(auto c:s)
         {
             if(c == ' ')
             {
                 space++;
             }
             else if(c >= '0' && c <= '9')
             {
                 num++;
             }
             else if(c >= 'a' && c <= 'z')
             {
                 ch++;
             }
             else if(c >= 'A' && c <= 'Z')
             {
                 ch++;
             }
             else
             {
                 other++;
             }
         }
         cout << ch << endl;
         cout << space << endl;
         cout << num << endl;
         cout << other << endl;
     }
     return 0;
 }

File: HJ40 统计字符.py
while True:
    try:
        sentence = input()
        word, space, digit, other = 0, 0, 0, 0
        for i in sentence:
            if i.isalpha():
                word += 1
            elif i == ' ':
                space += 1
            elif i.isdigit():
                digit += 1
            else:
                other += 1
        print(word)
        print(space)
        print(digit)
        print(other)
    except:
        break
